clang/lib/CIR/CodeGen/CIRGenDecl.cpp

```diff
@@ -660,13 +661,34 @@ CIRGenFunction::addInitializerToStaticVarDecl(const VarDecl &varDecl,
   if (globalOp.getSymType() != typedInit.getType()) {
     globalOp.setSymType(typedInit.getType());
 
+    cir::GlobalOp oldGlobalOp = globalOp;
+    globalOp =
+        builder.createGlobal(CGM.getModule(), getLoc(varDecl.getSourceRange()),
+                             oldGlobalOp.getName(), typedInit.getType(),
+                             oldGlobalOp.getConstant(), globalOp.getLinkage());
+    // FIXME(cir): OG codegen inserts new GV before old one, we probably don't
+    // need that?
```

Comment:
LLVM translation usually put them out in the proper order, because of the way MLIR handles this I don't think it should be a problem for us.